.github/workflows/ecosystem-ci-trigger.yml

```diff
@@ -9,7 +9,143 @@ jobs:
+              repo: context.repo.repo,
+              issue_number: prData.num,
+              labels: ['trigger: preview']
+            })
+            console.log('Added "trigger: preview" label.')
+
+      - name: Wait for Preview Release Completion (if Package Not Found)
+        if: steps.check-package.outputs.exists == false
+        uses: actions/github-script@v7
+        id: wait-preview-release
+        with:
+          script: |
+            const prData = ${{ steps.get-pr-data.outputs.result }}
+            const reaction = ${{ steps.check-package.outputs.reaction }}
+            const workflowFileName = 'preview-release.yml'
+            const workflow = await github.rest.actions.getWorkflow({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              workflow_id: workflowFileName,
+            })
+            const workflowId = workflow.data.id
+            console.log(`Waiting for workflow ID ${workflowId} to complete...`)
+
+            const maxRetries = 60 // Wait up to 10 minutes
+            const delay = 10000 // 10 seconds
+            let completed = false
+
+            for (let i = 0; i < maxRetries; i++) {
+              const runsData = await github.rest.actions.listWorkflowRuns({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                workflow_id: workflowId,
+                per_page: 100,
+                page: 1,
+              })
+
+              const runs = runsData.data.workflow_runs.filter(
+                run => run.head_sha === prData.commit
+              )
+
+              if (runs.length > 0) {
+                const latestRun = runs[0]
+                console.log(`Latest run status: ${latestRun.status}, conclusion: ${latestRun.conclusion}`)
+                if (latestRun.status === 'completed') {
+                  if (latestRun.conclusion === 'success') {
+                    console.log('Preview release workflow completed successfully.')
+                    completed = true
+                    break
+                  } else {
+                    throw new Error('Preview Release workflow failed.')
+                  }
+                }
+              }
+
+              console.log(`Retrying... (${i + 1}/${maxRetries})`)
+              await new Promise(resolve => setTimeout(resolve, delay))
+            }
+
+            if (!completed) {
+              throw new Error('Preview Release workflow did not complete in time.')
+            }
+
+            // Remove the 'rocket' reaction
+            if (reaction) {
+              await github.rest.reactions.deleteForIssueComment({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                comment_id: context.payload.comment.id, 
+                reaction_id: reaction,
+              })
+              console.log('Removed "rocket" reaction.')
+            }
+
+      - name: Check User Permissions
```

Comment:
Let's move this step and the "React Based on Permissions" step to the top so that nothing other than the reaction would happen for users without permission.